sdk/test/opentelemetry/sdk/trace/span_test.rb

```diff
@@ -463,6 +463,33 @@
     end
   end
 
+  describe '#instrumentation_library' do
+    it 'is identical to the instrumentation_scope' do
+      mock_span_processor.expect(:on_start, nil) { |s| yielded_span = s } # rubocop:disable Lint/UselessAssignment
+      span = Span.new(
+        context,
+        Context.empty,
+        OpenTelemetry::Trace::Span::INVALID,
+        'name',
+        SpanKind::INTERNAL,
+        nil,
+        span_limits,
+        [mock_span_processor],
+        nil,
+        nil,
+        Time.now,
+        nil,
+        OpenTelemetry::SDK::InstrumentationScope.new('foo', '1.0')
+      )
+
+      _(span.instrumentation_scope).must_be_instance_of(OpenTelemetry::SDK::InstrumentationScope)
+      _(span.instrumentation_library).must_be_instance_of(OpenTelemetry::SDK::InstrumentationLibrary)
+
```

Comment:
I don't think we should explicitly assert the types. Users should rely on duck-typing only.

```suggestion
```